### PR TITLE
memoize generateColorPalette and use palette for getColorTint

### PR DESCRIPTION
**Pull request by ethanshar**

### CODE REVIEW

File: src/style/colors.js [hunk 1]
```diff
@@ -99,13 +99,15 @@ class Colors {
...
-    let tintLevel = Math.floor(Number(tintKey) / 10 + 1);
-    tintLevel = Math.max(2, tintLevel);
-    tintLevel = Math.min(9, tintLevel);
-    return generateColorTint(color, tintLevel * 10);
+    let tintLevel = Math.floor(Number(tintKey) / 10);
+    tintLevel = Math.max(1, tintLevel);
+    tintLevel = Math.min(8, tintLevel);
+
+    const colorsPalette = this.generateColorPalette(color);
+    return colorsPalette[tintLevel - 1];
```
Comment (Inbal-Tish) [inline src/style/colors.js]: This returns the same color, not the tinted color.

File: src/style/colors.js [hunk 2]
```diff
@@ -130,7 +132,7 @@ class Colors {
 
     const sliced = tints.slice(0, 8);
     return sliced;
-  }
+  })
```
Comment (Inbal-Tish) [inline src/style/colors.js]: semicolon missing

File: src/style/colors.js (hunk 1 above)
Comment (ethanshar) [reply src/style/colors.js]: are you sure? 
there's a test that shows it works (: 
https://github.com/wix/react-native-ui-lib/blob/master/src/style/__tests__/colors.spec.js#L81

File: src/style/colors.js (hunk 2 above)
Comment (ethanshar) [reply src/style/colors.js]: done

File: src/style/colors.js (hunk 1 above)
Comment (Inbal-Tish) [reply src/style/colors.js]: I've called it and printed it. I keep getting the same color I sent

Comment (Inbal-Tish) [reply src/style/colors.js]: Ok. My bad. Seems to work

**Pull request merged**